### PR TITLE
Use tokio and futures non-alpha releases

### PR DESCRIPTION
**Pull request by dbcfd**
## What is the goal of this pull request?
Use the non-alpha versions of tokio and futures.

## What does this pull request change?
Dependencies and impacted code.

## What work remains to be done?
From<TypedData> is missing some fields.

## Do you consider it adequately tested?

YES

## Notes

### CODE REVIEW

File: azure-functions/src/generic.rs [hunk 1]
```diff
@@ -41,6 +41,10 @@ impl From<TypedData> for Value {
     fn from(data: TypedData) -> Self {
         match data.data {
             None => Value::None,
+            Some(Data::CollectionBytes(_c)) => unimplemented!(),
```
Comment (dbcfd) [inline azure-functions/src/generic.rs]: How should this be represented? Does value need additional types?

File: azure-functions-durable/src/client.rs [hunk 2]
```diff
@@ -93,6 +94,21 @@ pub struct Client {
     client: hyper::Client<hyper::client::HttpConnector>,
 }
 
+async fn body_from_response(res: hyper::Response<hyper::Body>) -> Result<Vec<u8>> {
+    let mut body = vec![];
+    let mut res = res.into_body();
+    while let Some(chunk) = res.next().await {
+        let chunk = chunk.map_err(|e| {
+            ClientError::Message(format!(
+                "failed to get read response: {}",
```
Comment (peterhuene) [inline azure-functions-durable/src/client.rs]: Nit: 

```suggestion
                "failed to read response: {}",
```

File: azure-functions-durable/src/client.rs [hunk 3]
```diff
@@ -93,6 +94,21 @@ pub struct Client {
     client: hyper::Client<hyper::client::HttpConnector>,
 }
 
+async fn body_from_response(res: hyper::Response<hyper::Body>) -> Result<Vec<u8>> {
```
Comment (peterhuene) [inline azure-functions-durable/src/client.rs]: Was this added over using `try_concat` for efficiency?

File: azure-functions-shared/Cargo.toml [hunk 4]
```diff
@@ -10,7 +10,7 @@ edition = "2018"
 
 [dependencies]
 azure-functions-shared-codegen = { version = "0.11.0", path = "../azure-functions-shared-codegen" }
-tonic = "0.1.0-alpha.5"
+tonic = { git = "https://github.com/hyperium/tonic.git", branch = "master" }
```
Comment (peterhuene) [inline azure-functions-shared/Cargo.toml]: We won't be able to take this PR until `tonic` publishes a needed version.  We've run into problems in the past of doing lots of work on top of git-referenced crates and then being at their mercy to release before we can publish new crates.

File: azure-functions/src/generic.rs (hunk 1 above)
Comment (peterhuene) [reply azure-functions/src/generic.rs]: I actually have this implemented in a WIP branch where I'm implementing Durable Functions 2.0.

Is updating the protobuf necessary for this PR or can it wait for that feature work?

**Comment by peterhuene** [comment]
Hi @dbcfd.

Thanks for the PR!  Would you mind rebasing this on `dev` and changing the target branch to `dev`?  This repo follows the branch semantics that the Azure Functions team uses where active development takes place in `dev` and `master` represents the current release.  It's annoying, but GitHub doesn't have the option to show `master` by default but to target `dev` with new PRs (afaik).

This would be the primary reason that CI is failing because the fix to CI exists in `dev` and not `master` (which only builds every release).

**Comment by dbcfd** [comment]
Yep, will rebase onto dev.

File: azure-functions/src/generic.rs (hunk 1 above)
Comment (dbcfd) [reply azure-functions/src/generic.rs]: Can wait for that feature work.

File: azure-functions-durable/src/client.rs (hunk 3 above)
Comment (dbcfd) [reply azure-functions-durable/src/client.rs]: More for getting types to line up. try_concat takes some fun type gymnastics. Can switch to that if that is preferred.

File: azure-functions-shared/Cargo.toml (hunk 4 above)
Comment (dbcfd) [reply azure-functions-shared/Cargo.toml]: Understand completely. I don't think a new version is too far off

File: azure-functions-durable/src/client.rs (hunk 3 above)
Comment (peterhuene) [reply azure-functions-durable/src/client.rs]: No preference.  Your implementation looks good to me.

**Comment by peterhuene** [comment]
@cla-bot check

File: azure-functions-shared/Cargo.toml (hunk 4 above)
Comment (peterhuene) [reply azure-functions-shared/Cargo.toml]: It looks like Tonic `0.1.0-beta.1` was released recently with a dependency of a non-preview 0.3.0 futures.  We have a update PR for Tonic that's failing to build, so I'm going to try to bump it to fix that PR.

**Comment by dbcfd** [comment]
Rebased against dev branch now.

**Comment by peterhuene** [comment]
So now we have a new blocker as we just took a dependency on `azure_sdk_core` for building the examples, which is using a preview version of hyper.

**Comment by peterhuene** [comment]
I think I'll feature-gate the example for now so that we can move forward on this PR.  I'll put up a fix.

**Comment by peterhuene** [comment]
Ok, if you rebase off of `dev` one more time (we need to delete that commit to the README since that incorrectly came from master) and update to the latest published tonic version, I think this will be good to go.

@dbcfd when you get a chance, could you open a PR against master to add yourself to the `.clabot` file at the root to "sign" the CLA (it indicates your contributions abide by the `LICENSE` for the repository).  That should get the cla-bot CI check happy once I merge it into master.

Thanks!

**Comment by peterhuene** [comment]
@cla-bot check

**Comment by dbcfd** [comment]
All libs updated again, using released tonic and updated prost as well.

**Comment by peterhuene** [comment]
@dbcfd thanks so much for pushing this PR through!

**Comment by peterhuene** [comment]
@all-contributors please add @dbcfd for code and maintenance

**Pull request merged**